amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -974,6 +973,36 @@ public void testErrorHandlerForUndeliverableAsyncTaskExceptions() {
         verify(rejectedTaskEvent, times(1)).accept(any());
     }
 
+    @Test
+    public void testUpdateStreamMapIfMissingLatestStream() throws Exception {
+        final List<StreamConfig> streamConfigList = createDummyStreamConfigList(1, 6);
+        retrievalConfig = new RetrievalConfig(kinesisClient, multiStreamTracker, applicationName)
+                .retrievalFactory(retrievalFactory);
+        scheduler = spy(new Scheduler(checkpointConfig, coordinatorConfig, leaseManagementConfig, lifecycleConfig,
+                metricsConfig, processorConfig, retrievalConfig));
+        when(multiStreamTracker.streamConfigList()).thenReturn(streamConfigList);
+        when(scheduler.shouldSyncStreamsNow()).thenReturn(true);
+
+        scheduler.checkAndSyncStreamShardsAndLeases();
+        verify(scheduler).syncStreamsFromLeaseTable(any());
+    }
+
+    @Test
+    public void testNotUpdatingStreamMapAsItContainsAllStreams() throws Exception {
+        final List<StreamConfig> streamConfigList = createDummyStreamConfigList(1, 6);
+        retrievalConfig = new RetrievalConfig(kinesisClient, multiStreamTracker, applicationName)
+                .retrievalFactory(retrievalFactory);
+        scheduler = spy(new Scheduler(checkpointConfig, coordinatorConfig, leaseManagementConfig, lifecycleConfig,
+                metricsConfig, processorConfig, retrievalConfig));
+        when(multiStreamTracker.streamConfigList()).thenReturn(streamConfigList);
+        when(scheduler.shouldSyncStreamsNow()).thenReturn(true);
+        // Populate currentStreamConfigMap to simulate that the leader has the latest streams.
+        streamConfigList.forEach(s -> scheduler.currentStreamConfigMap().put(s.streamIdentifier(), s));
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). With the exception of this line and the `VerificationMode` (line ~1003), this test is identical to the one above it. Create a reusable method.